test-driver/src/ComparisonContext.ts

```diff
@@ -50,16 +50,31 @@ export class ComparisonContext {
         return fs.readFileSync(path, options);
     }
 
-    setEventStream(ionReader: Reader) {
-        let eventStream = new IonEventStream(ionReader);
+    setEventStream(ionReader: Reader, path: string, args: IonCompareArgs) {
         let events: IonEvent[] = [];
+        try{
+            let eventStream = new IonEventStream(ionReader);
 
-        if(!eventStream.isEventStream) {
-            // processes input stream(text or binary) into event stream
-            events = this.collectInputStreamEvents(events, eventStream);
-            eventStream.events = events;
+            if(!eventStream.isEventStream) {
+                // processes input stream(text or binary) into event stream
+                events = this.collectInputStreamEvents(events, eventStream);
+                eventStream.events = events;
+            }
+            this.eventStream = eventStream;
+        } catch (EventStreamError) {
+            if(EventStreamError.eventstream) {
+                events.push(...EventStreamError.eventstream);
+            }
+            if(EventStreamError.type === "READ") {
```

Comment:
If we can't use the `ErrorType` enum for `READ` and `WRITE`, we should still try to have constants for those strings that we can refer to. I know this is a "where should it live?" question, and you'll have more context on this than I do, but we should try to find a home for it.

Reply:
Yes, I will add constants for these strings.